Antidote Verifier/Sdl.Community.AntidoteVerifier/Sdl.Community.AntidoteVerifier/Utils/CustomTextCollectionVisitor.cs

```diff
@@ -0,0 +1,333 @@
+﻿using Sdl.FileTypeSupport.Framework.BilingualApi;
+using System;
+using System.Collections.ObjectModel;
+using System.Text;
+
+namespace Sdl.Community.AntidoteVerifier
+{
+    public class CustomTextCollectionVisitor : IMarkupDataVisitor
```

Comment:
The following issues were noted with the connector 0.8.2 :
(examples are available on request)

1. Antidote was not able to correct the character starting a segment or following a formatting tag.
2. Antidote was not able to add a character before a tag-marker.
3. Antidote was not able to replace text containing a formatting tag.
4. The connector didn’t send text of reference tags to Antidote (but only formatting tags and locking tags).
5. Antidote didn’t know if a part of the text was read-only.

The connector 0.8.2 used a double approach : one for obtaining the text (via class TextCollectionVisitor), and a different one to replace the text (via the function GetCharacterCountingIterator). In order to easily address the above issues, it was simpler to use a unify approach for both obtaining and replacing the text.

Notes: 
- The update No. 5 for Antidote 9v4 is required to test the modifications for the above issues.
- As it is explained below, the modified connector handles tags : it separates tag-makers from tag-contents, and sends only tag-contents to Antidote.